### ISSUE TITLE
Guard performance...

### ISSUE THREAD

**Issue by Hypercubed** [pelotom/runtypes#95]
I've found that guards are very slow compared to a base implementation.  In fact, I've seen significant improvement by overriding the guards like:

```ts
String.guard = (x: unknown): x is string => {
  return typeof x === 'string';
};
```

I assume this is due to `guard` calling `validate` which is a try/catch wrapping the `check` function.  Have you thought about reversing the order here so that the `check` function is derived from a fast `guard` instead of the other way around?

Effectively I think this will mean changing the `create` function to be something like:

```ts
export function create<A extends Runtype>(guard: (x: {}) => x is Static<A>, A: any): A;
``` 
instead of:

```ts
export function create<A extends Runtype>(check: (x: {}) => Static<A>, A: any): A;
```

Additionally (or alternatively) could we allow defining the methods of the derived object in the A param:

```ts
export const Number = create<Number>(
  x => { ... }
  { 
    tag: 'number', 
    guard:  (x: unknown): x is string => {
      return typeof x === 'string';
    }
  }
);
```

**Comment by pelotom** [comment]
> I assume this is due to `guard` calling `validate` which is a try/catch wrapping the `check` function. Have you thought about reversing the order here so that the `check` function is derived from a fast `guard` instead of the other way around?

Yeah, others have made this observation and I agree it would be better to build on a validation object rather than exceptions. PRs welcome!

**Comment by pelotom** [comment]
I believe this is resolved now thanks to #99.
